share/models/change.py

```diff
@@ -177,6 +177,8 @@ def _create(self, save=True):
             except IntegrityError as e:
                 from share.disambiguation import disambiguate
                 logger.info('Handling unique violation error %r', e)
+                if self.target_type.model == 'identifier':
```

Comment:
I agree it's unavoidable, but the way we handle it will always result in an identifier pointing to multiple objects, and often at least one relation pointing to a "stub" instead of the work it actually relates to. That doesn't seem acceptable, and rewinding a few `MakeJsonPatches` tasks seemed better than trying to re-disambiguate the identified work/person at this stage. 

I could resolve the race condition without the task actually failing, though, by catching this particular exception in `MakeJsonPatches`, so it can retry more gracefully...